domaincommon/src/state.rs

```diff
@@ -359,9 +429,18 @@ impl Match {
         state: &<Match as Aggregate>::State,
         mech: &str,
         dir: &GridDirection,
+        cmd: &MechCommand,
     ) -> Result<Vec<<Match as Aggregate>::Event>> {
-        let mut evts = Vec::new();
         MatchState::validate_has_mech(state, mech)?;
+        if MatchState::validate_can_take_action(state, mech, cmd).is_err() {
```

Comment:
This same set of code occurs on line 357 and again elsewhere. We should probably have this set of code occur only once, above the `match` statement in the root command handler, since if we can't take an action, we don't need to ever enter the command-specific functions.